### PR TITLE
[feature](nereids) support multi_distinct_collect_list and multi_distinct_array_agg

### LINKED ISSUE #65244: [Enhancement] Support multiple DISTINCT aggregates for collect_list and array_agg
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

When a single GROUP BY contains more than one DISTINCT aggregate and any of them
is `collect_list(distinct ...)` or `array_agg(distinct ...)`, planning fails with:

    errCode = 2, ... can't support multi distinct.

`CheckMultiDistinct` rejects any distinct aggregate that does not implement
`SupportMultiDistinct`, and `collect_list` / `array_agg` do not implement it.
So queries like these cannot run:

    select g, collect_list(distinct a), collect_list(distinct b) from t group by g;
    select g, array_agg(distinct a), array_agg(distinct b) from t group by g;

Other aggregates (count, sum, sum0, group_concat) already support this through
the multi-distinct framework. `collect_list` and `array_agg` should too, via new
`multi_distinct_collect_list` and `multi_distinct_array_agg` functions.

### Solution

Add `multi_distinct_collect_list` / `multi_distinct_array_agg` and make
`CollectList` / `ArrayAgg` implement `SupportMultiDistinct`, plugging into the
existing multi-distinct rewrite (BE + FE).

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Baymine**
### What problem does this PR solve?

Issue Number: close #65244

Problem Summary:
When a single GROUP BY contained more than one DISTINCT aggregate and any of
them was collect_list(distinct ...) or array_agg(distinct ...), planning failed
with "... can't support multi distinct". CheckMultiDistinct rejects any distinct
aggregate that is not a SupportMultiDistinct, and collect_list / array_agg did
not implement that interface, so queries such as

    select g, collect_list(distinct a), collect_list(distinct b) from t group by g
    select g, array_agg(distinct a), array_agg(distinct b) from t group by g

could not run at all.

This adds the missing multi-distinct variants and wires them into the existing
framework:
- FE: CollectList and ArrayAgg now implement SupportMultiDistinct and convert to
  the new MultiDistinctCollectList / MultiDistinctArrayAgg functions. The new
  functions implement MultiDistinction, are registered in
  BuiltinAggregateFunctions, and get visitor hooks in AggregateFunctionVisitor.
- BE: registers multi_distinct_collect_list and multi_distinct_array_agg. Both
  dedup in the aggregate state via the existing Set-backed collect data, because
  the Nereids multi-distinct plan does not push the distinct argument into the
  group-by key.

### Release note

Support multiple DISTINCT aggregates in one GROUP BY when they include
collect_list(distinct ...) or array_agg(distinct ...).

### Check List (For Author)

- Test:
    - [x] Unit Test: MultiDistinctArrayAggTest, MultiDistinctCollectListTest, CheckMultiDistinctTest (28 cases, all pass).
    - [x] Regression test: added regression-test/suites/nereids_p0/multi_distinct/multi_distinct_collect_list.groovy (runs in CI).
- Behavior changed:
    - [x] Yes. Queries that previously failed with "can't support multi distinct" for collect_list/array_agg now plan and run.
- Does this need documentation?
    - [x] No.

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by Baymine** [comment]
run buildall

**Comment by hello-stephen** [comment]
# FE UT Coverage Report
Increment line coverage `  94.44% (34/36)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/65245_c0b5abda4eb3678bc2f1f91e86f6f811162dafd5/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/65245_c0b5abda4eb3678bc2f1f91e86f6f811162dafd5/fe_report/index.html)

File: be/src/exprs/aggregate/aggregate_function_array_agg.cpp [hunk 1]
```diff
@@ -25,9 +27,19 @@
 namespace doris {
 
 template <PrimitiveType T>
-AggregateFunctionPtr do_create_agg_function_collect(const DataTypes& argument_types,
+AggregateFunctionPtr do_create_agg_function_collect(bool distinct, const DataTypes& argument_types,
                                                     const bool result_is_nullable,
```
Comment (morrySnow) [inline be/src/exprs/aggregate/aggregate_function_array_agg.cpp:31]: **🐛 Null elements are silently dropped in the multi-distinct path**

The distinct path creates `AggregateFunctionCollect<AggregateFunctionCollectSetData<T, false>, false>` via `creator_without_type::create`. This factory wraps the inner function in `AggregateFunctionNullUnaryInline` (see `aggregate_function_null.h:331-342`), which **skips null rows** and never calls the inner `add()`. Additionally, `AggregateFunctionCollectSetData` uses `flat_hash_set<ElementType>` with no null map — it cannot represent null elements.

In contrast, the non-distinct array_agg path uses `AggregateFunctionArrayAggData` which explicitly tracks nulls via a null_map.

**Concrete failure:**
```sql
SELECT array_agg(distinct col) FROM (
  SELECT CAST(NULL AS INT) AS col UNION ALL SELECT 1 AS col
) t
```
Returns `[1]` instead of `[null, 1]`.

The behavior change is silent — it only triggers when multiple distinct aggregates cause the multi-distinct rewrite to fire.

**Suggested fix:** The multi_distinct_array_agg path needs a data structure that combines hash-set dedup with null tracking, similar to what `AggregateFunctionArrayAggData` does.

File: be/src/exprs/aggregate/aggregate_function_array_agg.cpp [hunk 2]
```diff
@@ -17,6 +17,8 @@
 
 #include "exprs/aggregate/aggregate_function_array_agg.h"
 
+#include "common/exception.h"
```
Comment (morrySnow) [inline be/src/exprs/aggregate/aggregate_function_array_agg.cpp:20]: **🧹 Unused include**

`#include "common/status.h"` is added but `Status` is never used in this file. Only `Exception` (from `"common/exception.h"`) is thrown in the new distinct path. Consider removing this include to follow include-what-you-use hygiene.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/CollectList.java [hunk 3]
```diff
@@ -102,4 +102,15 @@ public List<FunctionSignature> getSignatures() {
     public Expression resultForEmptyInput() {
         return new ArrayLiteral(new ArrayList<>(), this.getDataType());
     }
+
+    @Override
+    public AggregateFunction convertToMultiDistinct() {
+        Preconditions.checkArgument(distinct,
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/CollectList.java:110]: **⚠️ Limit argument silently dropped**

When `CollectList` is called with 2 children (the `collect_list(expr, limit)` variant), the limit argument `children.get(1)` is silently discarded. Only `children.get(0)` is passed to `MultiDistinctCollectList`. The limit parameter has no effect in the multi-distinct path.

Consider either:
1. Documenting that the limit is intentionally dropped in the multi-distinct path, or
2. Simplifying to always use `child(0)` since the `children.size() == 1` vs else branch both result in the same effective behavior.

**Review by morrySnow — CHANGES_REQUESTED** [review]
## Review Summary

This PR adds `multi_distinct_collect_list` and `multi_distinct_array_agg` support — a solid feature enabling queries with multiple DISTINCT aggregates in one GROUP BY that include collect_list/array_agg. The overall approach (FE: SupportMultiDistinct + MultiDistinction classes; BE: registering new function names) follows existing patterns. Good test coverage.

Below are the issues found, ranked by severity:

### 1. [HIGH] array_agg distinct path silently drops nulls
See inline comment in `aggregate_function_array_agg.cpp`. The Set-based dedup data structure cannot represent null elements, and the null-skipping adapter drops them before the inner function sees them. This means `array_agg(distinct nullable_col)` returns different results when it takes the multi-distinct path vs. the single-distinct path.

### 2. [MEDIUM] canSkewRewrite can produce nested ARRAY<ARRAY<T>>
Adding `SupportMultiDistinct` to ArrayAgg/CollectList makes `canSkewRewrite()` (`Aggregate.java:~206`) return true. In `SplitAggMultiPhase.java:~334`, phase 3 calls `getAggregateFunction()` which falls to the else branch: `aggFunc.withDistinctAndChildren(false, children)`. This wraps the already-array intermediate slot in another `array_agg()`, producing `ARRAY<ARRAY<T>>`. A query with a skew hint on these functions would return deeply nested arrays. Consider adding a guard in `canSkewRewrite()` or handling array concatenation in the skew phases.

### 3. [MEDIUM] DistinctWindowExpression doesn't handle ArrayAgg/CollectList
`DistinctWindowExpression.java:87-101` has hardcoded `instanceof` checks for Count, Sum, GroupConcat — it never queries `SupportMultiDistinct`. `ARRAY_AGG(DISTINCT col) OVER(...)` or `COLLECT_LIST(DISTINCT col) OVER(...)` would not be converted. Consider refactoring to use the `SupportMultiDistinct` interface like other rules do.

### 4. [LOW] get_name() returns "collect_set" for multi_distinct_* functions
`aggregate_function_collect.h:410-416`: `get_name()` returns "collect_set" for all Set-based data. Three different functions (`collect_set`, `multi_distinct_collect_list`, `multi_distinct_array_agg`) all report as "collect_set" in EXPLAIN ANALYZE profiles and error messages, making debugging confusing.

### 5. [LOW] Unused #include
See inline comment in `aggregate_function_array_agg.cpp`.

### 6. [LOW] convertToMultiDistinct silently drops limit argument
See inline comment in `CollectList.java`.

**Comment by hello-stephen** [comment]
# BE Regression && UT Coverage Report
Increment line coverage `88.89% (16/18)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/65245_c0b5abda4eb3678bc2f1f91e86f6f811162dafd5_merge/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/65245_c0b5abda4eb3678bc2f1f91e86f6f811162dafd5_merge/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 73.02% (29143/39913) |
| Line Coverage     | 56.63% (315018/556315) |
| Region Coverage   | 53.05% (261883/493638) |
| Branch Coverage   | 54.16% (115158/212637) |

**Comment by hello-stephen** [comment]
# FE Regression Coverage Report
Increment line coverage `  35.48% (22/62)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/65245_c0b5abda4eb3678bc2f1f91e86f6f811162dafd5_merge_fe/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/65245_c0b5abda4eb3678bc2f1f91e86f6f811162dafd5_merge_fe/report/index.html)

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/CollectList.java (hunk 3 above)
Comment (Baymine) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/CollectList.java:110]: the else branch does pass children.get(1), so the limit is preserved and MultiDistinctCollectList has a matching 2-arg signature.

File: regression-test/suites/nereids_p0/multi_distinct/multi_distinct_collect_list.groovy [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    // (b) DISTINCT dedups. Two distinct args (a AND b) exercise the multi-distinct path (getDistinctArguments().size() > 1); array_sort() makes element order deterministic.
+    def cl = sql """
+        select g,
+               array_sort(collect_list(distinct a)) la,
+               array_sort(collect_list(distinct b)) lb
+        from multi_distinct_cl
+        group by g
+        order by g
+    """
+    assertEquals(2, cl.size())
```
Comment (feiniaofeiafei) [inline regression-test/suites/nereids_p0/multi_distinct/multi_distinct_collect_list.groovy:65]: **P2 — This result check does not follow the regression-test convention.** Deterministic expected results should use `qt_`/`order_qt_` and a generated `.out` file instead of manual `assertEquals` checks. Please convert these result assertions accordingly. The test data is also entirely non-null, so it misses the `array_agg(DISTINCT nullable_col)` semantic regression above; please add nullable input and the supported/unsupported type boundary cases.

File: be/src/exprs/aggregate/aggregate_function_array_agg.cpp [hunk 5]
```diff
@@ -44,23 +56,25 @@ AggregateFunctionPtr create_aggregate_function_array_agg(const std::string& name
...
-                                                              attr);
+        agg_fn = do_create_agg_function_collect<INVALID_TYPE>(distinct, argument_types,
+                                                              result_is_nullable, attr);
     }
     return agg_fn;
 }
 
 void register_aggregate_function_array_agg(AggregateFunctionSimpleFactory& factory) {
     factory.register_function_both("array_agg", create_aggregate_function_array_agg);
+    factory.register_function_both("multi_distinct_array_agg", create_aggregate_function_array_agg);
```
Comment (feiniaofeiafei) [inline be/src/exprs/aggregate/aggregate_function_array_agg.cpp:87]: **P3 — The registered function is reported as `collect_set` by BE.** This implementation instantiates `AggregateFunctionCollect<AggregateFunctionCollectSetData<...>>`, whose `get_name()` returns `collect_set`. Therefore `multi_distinct_array_agg` (and similarly `multi_distinct_collect_list`) is mislabeled in profiles and error/debug output. Please preserve the registered function identity, for example by passing/storing the logical function name or using a dedicated implementation whose `get_name()` returns the correct name.

File: be/src/exprs/aggregate/aggregate_function_array_agg.cpp [hunk 6]
```diff
@@ -25,9 +27,19 @@
...
-AggregateFunctionPtr do_create_agg_function_collect(const DataTypes& argument_types,
+AggregateFunctionPtr do_create_agg_function_collect(bool distinct, const DataTypes& argument_types,
                                                     const bool result_is_nullable,
                                                     const AggregateFunctionAttr& attr) {
+    if (distinct) {
+        if constexpr (T == INVALID_TYPE) {
+            throw Exception(ErrorCode::INTERNAL_ERROR,
+                            "unexpected type for array_agg distinct, please check the input");
+        } else {
+            return creator_without_type::create<
```
Comment (feiniaofeiafei) [inline be/src/exprs/aggregate/aggregate_function_array_agg.cpp:47]: **P1 — `array_agg(DISTINCT nullable_col)` drops NULL.** The normal nullable `array_agg` path uses `create_ignore_nullable` and lets `AggregateFunctionArrayAggData` preserve the null map. This new set-based path uses the regular nullable adapter, which skips null input rows before the inner aggregate sees them, and `AggregateFunctionCollectSetData` has no state for a distinct NULL. As a result, the multi-distinct plan returns an array without NULL, while the existing single-distinct path should retain one NULL. Please preserve one distinct NULL explicitly and add a regression case with nullable input.

File: be/src/exprs/aggregate/aggregate_function_array_agg.cpp [hunk 7]
```diff
@@ -25,9 +27,19 @@
 namespace doris {
 
 template <PrimitiveType T>
-AggregateFunctionPtr do_create_agg_function_collect(const DataTypes& argument_types,
+AggregateFunctionPtr do_create_agg_function_collect(bool distinct, const DataTypes& argument_types,
                                                     const bool result_is_nullable,
                                                     const AggregateFunctionAttr& attr) {
+    if (distinct) {
+        if constexpr (T == INVALID_TYPE) {
```
Comment (feiniaofeiafei) [inline be/src/exprs/aggregate/aggregate_function_array_agg.cpp:34]: **P2 — FE and BE advertise different type support.** `MultiDistinctArrayAgg` declares an `AnyDataType` argument, so FE can accept types that `dispatch_switch_all` does not dispatch, but this branch turns them into an `INTERNAL_ERROR` at execution time. Please either implement the distinct state for every FE-advertised type or reject unsupported types during FE analysis with a user-facing error. An accepted query must not fail later with an internal error.

**Comment by Baymine** [comment]
run buildall

**Comment by hello-stephen** [comment]
# BE UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/2c642e4bb9e46bfe975acea417d69c241058b635_2c642e4bb9e46bfe975acea417d69c241058b635/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/2c642e4bb9e46bfe975acea417d69c241058b635_2c642e4bb9e46bfe975acea417d69c241058b635/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 56.53% (23196/41036) |
| Line Coverage     | 40.08% (225118/561713) |
| Region Coverage   | 36.03% (177743/493298) |
| Branch Coverage   | 37.08% (79000/213035) |

**Comment by Baymine** [comment]
run buildall

**Comment by hello-stephen** [comment]
<details>
<summary>TPC-H: <b>Total hot run time: 29647 ms</b></summary>

```
machine: 'aliyun_ecs.c7a.8xlarge_32C64G'
scripts: https://github.com/apache/doris/tree/master/tools/tpch-tools
Tpch sf100 test result on commit 39c337d0e3887ebe78faa4e308c0f018dfe57d38, data reload: false

------ Round 1 ----------------------------------
============================================
q1	17873	4082	4033	4033
q2	2027	330	199	199
q3	10262	1440	816	816
q4	4691	481	347	347
q5	7522	864	577	577
q6	204	182	139	139
q7	784	842	612	612
q8	10126	1573	1626	1573
q9	5925	4406	4391	4391
q10	6823	1798	1524	1524
q11	507	354	315	315
q12	755	576	441	441
q13	18133	3430	2723	2723
q14	277	271	257	257
q15	q16	785	785	717	717
q17	1052	1033	889	889
q18	7040	5792	5688	5688
q19	1454	1383	1099	1099
q20	739	644	560	560
q21	5699	2577	2453	2453
q22	431	351	294	294
Total cold run time: 103109 ms
Total hot run time: 29647 ms

----- Round 2, with runtime_filter_mode=off -----
============================================
q1	4383	4282	4283	4282
q2	286	340	214	214
q3	4663	5051	4394	4394
q4	2059	2166	1363	1363
q5	4456	4303	4360	4303
q6	238	182	128	128
q7	2144	1912	1632	1632
q8	2463	2176	2105	2105
q9	8007	7864	7834	7834
q10	4793	4773	4335	4335
q11	572	561	404	404
q12	751	798	541	541
q13	3290	3614	3017	3017
q14	309	311	270	270
q15	q16	717	741	668	668
q17	1356	1327	1310	1310
q18	8140	7624	7064	7064
q19	1091	1098	1055	1055
q20	2260	2230	1960	1960
q21	5215	4583	4464	4464
q22	501	458	420	420
Total cold run time: 57694 ms
Total hot run time: 51763 ms
```
</details>

**Comment by hello-stephen** [comment]
<details>
<summary>TPC-DS: <b>Total hot run time: 180004 ms</b></summary>

```
machine: 'aliyun_ecs.c7a.8xlarge_32C64G'
scripts: https://github.com/apache/doris/tree/master/tools/tpcds-tools
TPC-DS sf100 test result on commit 39c337d0e3887ebe78faa4e308c0f018dfe57d38, data reload: false

query5	4372	623	488	488
query6	474	212	192	192
query7	4966	591	342	342
query8	352	183	172	172
query9	8808	4035	4012	4012
query10	472	346	286	286
query11	5848	2364	2177	2177
query12	163	106	100	100
query13	1278	582	450	450
query14	6312	5280	4899	4899
query14_1	4333	4269	4249	4249
query15	212	202	179	179
query16	1005	463	432	432
query17	1124	709	572	572
query18	2434	470	342	342
query19	202	191	144	144
query20	108	104	103	103
query21	229	148	130	130
query22	13629	13675	13472	13472
query23	17420	16507	16249	16249
query23_1	16229	16231	16184	16184
query24	7517	1755	1287	1287
query24_1	1298	1305	1294	1294
query25	539	448	371	371
query26	1335	353	207	207
query27	2604	615	387	387
query28	4428	2040	2021	2021
query29	1108	626	495	495
query30	340	272	230	230
query31	1121	1084	972	972
query32	103	64	60	60
query33	540	323	271	271
query34	1162	1150	656	656
query35	781	799	686	686
query36	1438	1417	1253	1253
query37	154	107	95	95
query38	1877	1718	1653	1653
query39	926	931	877	877
query39_1	867	902	877	877
query40	255	167	142	142
query41	74	69	68	68
query42	100	95	97	95
query43	320	328	287	287
query44	1438	793	779	779
query45	201	198	183	183
query46	1048	1194	765	765
query47	2388	2330	2241	2241
query48	427	422	296	296
query49	584	433	318	318
query50	1110	425	332	332
query51	10635	10548	10868	10548
query52	88	87	78	78
query53	250	283	210	210
query54	314	246	237	237
query55	77	74	70	70
query56	320	315	290	290
query57	1452	1422	1364	1364
query58	314	272	264	264
query59	1589	1655	1418	1418
query60	340	275	262	262
query61	171	171	171	171
query62	702	651	576	576
query63	235	203	208	203
query64	2931	1032	866	866
query65	4826	4766	4757	4757
query66	1827	511	388	388
query67	28890	29494	29326	29326
query68	3265	1607	964	964
query69	411	303	271	271
query70	1036	966	980	966
query71	353	303	309	303
query72	3076	2692	2405	2405
query73	829	807	453	453
query74	5096	4950	4732	4732
query75	2617	2572	2205	2205
query76	2328	1179	809	809
query77	352	375	286	286
query78	12407	12414	11641	11641
query79	1467	1185	793	793
query80	1275	526	456	456
query81	524	326	285	285
query82	612	158	122	122
query83	359	312	287	287
query84	270	155	130	130
query85	963	634	495	495
query86	418	305	289	289
query87	1844	1829	1752	1752
query88	3748	2827	2786	2786
query89	458	409	353	353
query90	1903	201	193	193
query91	201	185	166	166
query92	61	59	56	56
query93	1701	1562	996	996
query94	743	347	299	299
query95	765	559	475	475
query96	1099	814	368	368
query97	2707	2668	2552	2552
query98	211	204	200	200
query99	1197	1173	1036	1036
Total cold run time: 265746 ms
Total hot run time: 180004 ms
```
</details>

**Comment by hello-stephen** [comment]
<details>
<summary>ClickBench: <b>Total hot run time: 25.09 s</b></summary>

```
machine: 'aliyun_ecs.c7a.8xlarge_32C64G'
scripts: https://github.com/apache/doris/tree/master/tools/clickbench-tools
ClickBench test result on commit 39c337d0e3887ebe78faa4e308c0f018dfe57d38, data reload: false

query1	0.00	0.00	0.00
query2	0.09	0.05	0.05
query3	0.25	0.14	0.13
query4	1.61	0.14	0.14
query5	0.28	0.22	0.23
query6	1.23	1.07	1.04
query7	0.04	0.01	0.00
query8	0.06	0.04	0.04
query9	0.39	0.31	0.32
query10	0.56	0.59	0.60
query11	0.20	0.14	0.14
query12	0.18	0.14	0.15
query13	0.48	0.48	0.48
query14	1.03	1.01	0.98
query15	0.61	0.59	0.60
query16	0.35	0.34	0.32
query17	1.08	1.11	1.12
query18	0.23	0.22	0.21
query19	2.04	1.95	1.96
query20	0.02	0.01	0.01
query21	15.45	0.21	0.13
query22	4.96	0.06	0.05
query23	16.10	0.32	0.12
query24	2.89	0.39	0.33
query25	0.12	0.04	0.03
query26	0.73	0.20	0.16
query27	0.04	0.04	0.05
query28	3.45	0.89	0.54
query29	12.47	4.10	3.28
query30	0.28	0.15	0.16
query31	2.79	0.60	0.31
query32	3.22	0.60	0.50
query33	3.20	3.27	3.20
query34	15.71	4.26	3.52
query35	3.54	3.49	3.52
query36	0.55	0.43	0.45
query37	0.09	0.07	0.06
query38	0.05	0.04	0.04
query39	0.04	0.03	0.03
query40	0.18	0.17	0.15
query41	0.09	0.03	0.02
query42	0.05	0.03	0.03
query43	0.04	0.03	0.04
Total cold run time: 96.77 s
Total hot run time: 25.09 s
```
</details>

**Comment by hello-stephen** [comment]
# FE Regression Coverage Report
Increment line coverage `  51.92% (27/52)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/65245_39c337d0e3887ebe78faa4e308c0f018dfe57d38_merge_fe/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/65245_39c337d0e3887ebe78faa4e308c0f018dfe57d38_merge_fe/report/index.html)

**Comment by hello-stephen** [comment]
# BE Regression && UT Coverage Report
Increment line coverage `96.92% (63/65)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/65245_39c337d0e3887ebe78faa4e308c0f018dfe57d38_merge/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/65245_39c337d0e3887ebe78faa4e308c0f018dfe57d38_merge/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 74.78% (30241/40440) |
| Line Coverage     | 58.83% (332336/564880) |
| Region Coverage   | 55.55% (278599/501500) |
| Branch Coverage   | 56.83% (123216/216824) |

**Comment by feiniaofeiafei** [comment]
The issue mentioned in this PR—where the two SQL statements select g, collect_list(distinct a), collect_list(distinct b) from t group by g and select g, array_agg(distinct a), array_agg(distinct b) from t group by g could not be executed—has been resolved after the merge of https://github.com/apache/doris/pull/65664. This PR implements a different approach to enable execution, using CTE (Common Table Expression) splitting to achieve the desired functionality.
